HIPs/baf.md

```diff
@@ -0,0 +1,248 @@
+* [Helm](https://helm.sh/) is a package manager for K8s. Helm Charts are configuration files designed for K8s to help define, install and upgrade complex K8s applications. BAF uses Helm Charts for designing and configuring the architecture of each DLT/Blockchain platform for its own network set-up.
+* [HashiCorp Vault](https://www.vaultproject.io/) provisions a secure approach to store and gain secret information such as tokens, passwords and certificates. BAF relies on Vaults for managing certificates used in each node of a DLT/Blockchain network during the lifecycle of a deployment, and it is a prerequisite that the Vault is installed and unsealed prior to deployment of a DLT/Blockchain network. Vault provides a cloud agnostic key management solution by default and can be swapped with other secure key management solutions if desired.
+* [GitOps](https://www.weave.works/technologies/gitops/) introduces an approach that can make K8s cluster management easier and also guarantee the latest application delivery is on time. BAF uses Weavework’s Flux for the implementation of GitOps and executes an Ansible role called setup/flux defined in the BAF GitHub repo.
+
+In addition to the above tools, we are of course dependent on the releases made by other DLT platform projects. The deployment of DLT networks currently supported by BAF are:
+* Hyperledger Fabric
+* Hyperledger Besu
+* Hyperledger Indy
+* GoQuorum
+* Corda Open Source
+* Corda Enterprise
+
+# Motivation
+<mark>_**Motivation** for this project, a longer justification of the project may be a couple of hundred words. Why is this project better at solving a problem compared to parallel proposals or implemented projects?_</mark>
+
+Setting up a new DLT/Blockchain network or maintaining an existing DLT/Blockchain network in a production-scale environment is not straightforward. For the existing DLT/Blockchain platforms, each has its own architecture, which means the same way of setting up one DLT/Blockchain network cannot be applied to others.
+
+Therefore, when blockchain developers are asked to use an unfamiliar DLT/Blockchain platform, it requires significant effort for even experienced technicians to properly setup the DLT/Blockchain network. This is especially true in large-scale production projects across heterogeneous corporate environments which require other key aspects such as security and service availability.
+
+Being aware of the potential difficulty and complexity of getting a production-scale DLT/Blockchain network ready, cloud vendors have provisioned their own managed Blockchain services (aka Blockchain as a Service or BaaS) to help alleviate various pain-points during the process. However, limitations can still be identified in these BaaS solutions (e.g., limited network size, locked to all nodes on a single cloud provider, limited choice of DLT/Blockchain platform).
```

Comment:
I think the motivation for BAF should not be painted as "an alternative to BaaS" or "addressing limitations of BaaS",  because BAF vs. BaaS are apples and oranges.

BAF certainly takes care of the provisioning of blockchain networks, but that's only a (small) part of the blockchain network lifecycle management, other aspects include:
- consortium governance
- ongoing software upgrades while the network is running, including complex orchestration procedures like hard forks (such as for Ethereum based networks)
- monitoring metrics (both low-level resources metrics as well as DLT-level consensus liveness checks)
- backup and recovery
- etc.

High quality BaaS platforms offer all the above, plus the support that nodes in the same blockchain network can be run either in the cloud or on-prem (including private data centers and dedicated cloud accounts).

It may be OK to assume that one or a few parties in a consortium are capable of provisioning and operating blockchain nodes themselves, it'd be a mistake to assume that all parties are capable of and willing to doing that. As the saying goes "you are as fast as your slowest party", BaaS based fully managed nodes must be considered as part of any enterprise consortium's deployment options.

I think it would be more accurate to say that BAF is complementary to BaaS. It'd be interesting to see what it would take to make BAF work together with some leading BaaS platforms so it can help deploy on-prem nodes.

Finally, some BaaS platforms are not subject to the cited limitations (limited network size, locked to all nodes on a single cloud provider, limited choice of DLT/Blockchain platform) at all. So that sentence should be clarified.

Reply:
Agree that the limitations are not inherent in all BaaS. But these reasons are the motivations for creating BAF. We would love to see some BaaS providers use BAF in their solutions.